peakinvestigator/peakinvestigator_saas.py

```diff
@@ -62,17 +66,22 @@ def execute(self, action):
         response = requests.post(self._server + "/api/", data=action.build_query())
         return response.text
         
-    def upload(self, sftp_action, local_file, remote_file, callback=None):
-        """Upload a file to a SFTP server."""
-        
-        with SSHClient() as ssh:
-            ssh.set_missing_host_key_policy(WarningPolicy())
-            ssh.connect(sftp_action.host, port=sftp_action.port,
-                            username=sftp_action.sftp_username,
-                            password=sftp_action.sftp_password)
-            with ssh.open_sftp() as sftp:
-                sftp.put(local_file, remote_file, callback)
+    def upload(self, upload_action, local_file, num_scan=0):
+        """Upload file to a SFTP server"""
+        upload_action.num = num_scan
+        uploader = Uploader(upload_action.host, upload_action.token, "Progress_Factory2")
```

Comment:
You've hard-coded `Progress_Factory2` here. How does a user of the SDK specify their own progress factory?

Reply:
I used this definition from earlier feedback, so progress_factory2 can be tqdm, progress_factory, or another progress GUI provided. However, I see your point. 
Instead of having uploader = Uploader(...,'progress_factory), should this be defined in the input script (example below)?

from peakinvestigator import PeakInvestigatorSaaS
from peakinvestigator.actions import *

service = PeakInvestigatorSaaS()
action = UploadAction("VERSION", "USERID", "PASSWRD","project_id","data_name") 
response = service.execute(action)
action.process_response(response)
call = service.upload(action,"file_name",'progress_gui',num_scan)